### PR TITLE
Feature/id 12/connect api to relational db

### PR DESCRIPTION
**Pull request by jhael-arce-jalafund**
# Pull Request Title
#### Resolved task/US: ([[ID-12] Database setup](https://trello.com/c/TWQ4Fva0/15-id-12-database-setup))
* * * *
image
image
image

**Execute :**
docker-compose up --build
dotnet ef migrations add InitialCreate
dotnet ef database update

## What's in this pull request?
This pull request implements the functionality to connect the API to a relational database using Entity Framework Core. It allows for CRUD operations on Expenses, Incomes, and Users through the API endpoints. The necessary code and mappings have been added to ensure that the API interacts with the database correctly, utilizing the provided Dockerfile for database setup.

Motivation and Context: The motivation for this change is to fulfill the acceptance criteria of the user story, ensuring that the API can persist data in a relational database and perform CRUD operations effectively.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style [guidelines](https://github.com/programacion-6/ExPenSeTraCkerGroUpSiX/wiki/Convetions) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: RestApi/Controllers/IncomesController.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+        [HttpDelete("{id:guid}")]
+        public async Task<IActionResult> Delete([FromRoute] Guid id)
+        {
+            var result = await _incomeService.DeleteAsync(id);
+            return result ? NoContent() : NotFound();
+        }
+
+
+    }
```
Comment (JosuePrado) [inline RestApi/Controllers/IncomesController.cs]: Missing endline

File: RestApi/Mappers/CreateIncomeRequest.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    public Income ToDomain()
+    {
+        return new Income
+        {
+            Amount = Amount,
+            Date = Date,
+            Source = Source
+        };
+    }
+}
```
Comment (JosuePrado) [inline RestApi/Mappers/CreateIncomeRequest.cs]: Missing endLine

File: RestApi/Controllers/UserController.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    {
+        var createdUser = await _userManager.CreateAsync(user);
+        return CreatedAtAction(nameof(Get), new { id = createdUser.Id }, createdUser);
+    }
+
+    [HttpPut("profile")]
+    public async Task<IActionResult> Update([FromBody] User user)
+    {
+        if (user.Id == Guid.Empty)
+            return BadRequest("El ID del usuario es obligatorio.");
```
Comment (JosuePrado) [inline RestApi/Controllers/UserController.cs]: can you put the message in English, pls

File: RestApi/Persistence/DataBase/DbConstants.cs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+namespace RestApi.Persistence.DataBase;
+
+public static class DbConstants
+{
+    public const string DefaultConnectionPath = "Database:ConnectionString:DefaultConnection";
+}
```
Comment (JosuePrado) [inline RestApi/Persistence/DataBase/DbConstants.cs]: Missing endLine

File: RestApi/Program.cs [hunk 5]
```diff
@@ -1,44 +1,32 @@
...
-.WithOpenApi();
+app.MapControllers();
 
-app.Run();
-
-record WeatherForecast(DateOnly Date, int TemperatureC, string? Summary)
-{
-    public int TemperatureF => 32 + (int)(TemperatureC / 0.5556);
-}
+app.Run();
```
Comment (JosuePrado) [inline RestApi/Program.cs]: Missing endline

File: RestApi/RequestPipeline/WebApplicationExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        app.Map("/error", (HttpContext httpContext) =>
+        {
+            var exception = httpContext.Features.Get<IExceptionHandlerFeature>()?.Error;
+
+            return Results.Problem();
+        });
+
+        return app;
+    }
+}
```
Comment (JosuePrado) [inline RestApi/RequestPipeline/WebApplicationExtensions.cs:28]: Missing endline

File: RestApi/RequestPipeline/WebApplicationExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+public static class WebApplicationExtensions
+{
+    public static WebApplication InitializeDatabase(this WebApplication app)
+    {
+        DbInitializer.Initialize(app.Configuration[DbConstants.DefaultConnectionPath]);
+
+        return app;
+    }
+
+    public static WebApplication UseGlobalErrorHandling(this WebApplication app)
```
Comment (JosuePrado) [inline RestApi/RequestPipeline/WebApplicationExtensions.cs:15]: what does this method do ?

File: RestApi/Persistence/DataBase/ApplicationDbContext.cs [hunk 8]
```diff
@@ -0,0 +1,18 @@
+using Microsoft.EntityFrameworkCore;
+using RestApi.Domain;
+
+namespace RestApi.Persistence.DataBase;
+
+public class ApplicationDbContext : DbContext
+    {
+        public DbSet<User> Users { get; set; }
```
Comment (JosuePrado) [inline RestApi/Persistence/DataBase/ApplicationDbContext.cs:8]: This is not in accordance with the design of uml

File: RestApi/appsettings.json [hunk 9]
```diff
@@ -5,5 +5,8 @@
       "Microsoft.AspNetCore": "Warning"
     }
   },
+  "ConnectionStrings": {
+    "DefaultConnection": "Host=localhost;Port=5432;Database=helios;Username=phaeton;Password=ecliptic"
+  },
   "AllowedHosts": "*"
-}
+}
```
Comment (JosuePrado) [inline RestApi/appsettings.json]: Missing endline

File: RestApi/RequestPipeline/WebApplicationExtensions.cs (hunk 7 above)
Comment (jhael-arce-jalafund) [reply RestApi/RequestPipeline/WebApplicationExtensions.cs:15]: The UseGlobalErrorHandling method sets up global error handling. If an error occurs, the user is redirected to /error, where the exception is captured, and a generic error response (like 500) is returned, protecting internal details.

**Review by JosuePrado — APPROVED** [review]

**Pull request merged**